### DISCUSSION TITLE
IRTcummulative Example

### DISCUSSION THREAD

**Discussion by AbigailHemming** [topas-nbio/TOPAS-nBio#7] [Simulation-question] [answered]
### Contact Details

abigail.hemming@postgrad.manchester.ac.uk

### What was the issue?

[TsIRTCummulativecopy.txt](https://github.com/user-attachments/files/15873253/TsIRTCummulativecopy.txt)
[RADE-20-00084.1-1.pdf](https://github.com/user-attachments/files/15873445/RADE-20-00084.1-1.pdf)
[GaussianAllReactions_PlanteReactions_proton_5.0_MeV_0.5_Gy_25uM_FWHM_1.8us (copy).txt](https://github.com/user-attachments/files/15873499/GaussianAllReactions_PlanteReactions_proton_5.0_MeV_0.5_Gy_25uM_FWHM_1.8us.copy.txt)
![OH^0 v3 1](https://github.com/topas-nbio/TOPAS-nBio/assets/173077893/fd6d8cdc-9aad-4f31-8188-ed78e66ea38a)

Hi, 
I am attempting to use the TsIRTcummulative example to reproduce the results seen in the attached paper by Ramos-Méndez et al, specifically the figure 2 graph corresponding to 100MeV protons and a 1 ns pulse width, as a starting point for my simulations. I have noticed within figure 2 that the number of hydroxyl radicals at 10^-6 us is non zero however within my simulations the number starts at 0 increases to a peak and decreases- see attached graph. 
Is this due to the inclusion of hydroxyls produced in the pre-chemical stage? If so how can this be done using this example code?

### TOPAS-nBio category

Scoring

### Version

3.0

### What Operative System (OS) are you running on?

Linux (Debian/Ubuntu)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by tmasilela** [comment]
Hi Abigail.

If I've understood your question correctly you need to include the following line in your parameter file:

`b:Sc/gvalue/ForceLowTimeCutTo1ps = "True"`

I have attached the resulting [phase space](https://github.com/user-attachments/files/15875688/test.txt) that shows that the OH yields are now non-zero at 1 ps. Please le me know if this solves your issue.

All the best,

Thongchai

**Reply by AbigailHemming** [reply]
Hi, 
this has worked to make the OH yield non zero at 1ps however the plot produced still looks quite different to the 100MeV 1ns graph in figure 2 of the paper I attached, specifically in the early time period before a peak is reached- approx 1-10^3 ps. I also note that when plotted our phase spaces are quite different too wrt to the rate of energy deposition change and number of molecules, please see attached figures, tmasilela.png corresponds to your phase space where as the other corresponds to the phase space that results from running the attached parameter file (TsIRTCummulativecopy.txt) with the line b:Sc/gvalue/ForceLowTimeCutTo1ps = "True" added. 

Could you clarify whether any additional changes were made to the attached parameter file and why the output graphs are different to those in figure 2?

My plot:
![100MeV_1ns_0 5Gy](https://github.com/topas-nbio/TOPAS-nBio/assets/173077893/40c67a19-7d07-407f-85ee-b1a25bbd7618)
your plot:
![tmasilelaplot](https://github.com/topas-nbio/TOPAS-nBio/assets/173077893/8bd559b4-1453-4b82-8fff-4953dfece0ed)
Phase space for my plot: 
[GaussianAllReactions_PlanteReactions_proton_100.0_MeV_0.5_Gy_25uM_FWHM_1ns_forcetimestop.txt](https://github.com/user-attachments/files/15901816/GaussianAllReactions_PlanteReactions_proton_100.0_MeV_0.5_Gy_25uM_FWHM_1ns_forcetimestop.txt)
Parameter file: 
[TsIRTCummulativecopy.txt](https://github.com/user-attachments/files/15901825/TsIRTCummulativecopy.txt)
reference paper:
[RADE-20-00084.1-1-1.pdf](https://github.com/user-attachments/files/15901873/RADE-20-00084.1-1-1.pdf)

Thank you for your help!

**Reply by tmasilela** [reply]
Hi Abigail,

Sorry for the confusion. I was pressed for time so I changed the prescribed dose in order to get the simulation to end faster.

In terms of the discrepancy during the early times, this is due to the fact that you need to perform multiple runs. According to the referenced paper, results were averaged from 130 batches. I have pasted the results of my simulations below, but you can find the raw data+parameter files [here](https://ucsf.box.com/s/1vo6q8pv40pn5l4iexp49dm21yj49qqo).

![image](https://github.com/topas-nbio/TOPAS-nBio/assets/159175671/fd475b48-391c-4d5a-8990-724eccf5fb40)

Some things to note: I changed the source to "Uniform" to be more in line with the paper and consequently changed the pulse FWHM and Mean to 1 ns and .5 ns respectively.

As you can see, while this fixes the issues at early times, the OH doesn't drop as expected at ~100 us. This is likely due to a missing reaction/s in the simulation which scavenges OH at longer times. Unfortunately I wasn't able to pinpoint exactly which reaction at the moment, but I'll try get back to you when I have time.

Hope this was useful.

All the best,

Thongchai

**Comment by JoseRamosMendez** [comment, marked as answer]
It seems that you're missing several type VI reactions. You can try with the following file 
[PlanteTypeVI.txt](https://github.com/user-attachments/files/15975824/PlanteTypeVI.txt)
    
    includeFile = PlanteTypeVI.txt

Extend the time domain to `10^4 us`. And add this parameter to your main TOPAS-nBio parameter file.

    b:Sc/gvalue/EnableHighTimeScavengers = "True"


![Screenshot 2024-06-25 at 11 09 34 AM](https://github.com/topas-nbio/TOPAS-nBio/assets/12536336/14a27db2-9010-4180-ab3b-52cdcd9e3b1d)
